core/src/avm2/globals/xml_list.rs

```diff
@@ -1,4 +1,77 @@
 //! XMLList builtin and prototype
 
-// XMLList currently uses the same instance allocator as XML
-pub use crate::avm2::object::xml_allocator as xml_list_allocator;
+pub use crate::avm2::object::xml_list_allocator;
+use crate::avm2::{
+    e4x::{simple_content_to_string, E4XNode, E4XNodeKind},
+    object::E4XOrXml,
+    Activation, Error, Object, TObject, Value,
+};
+
+fn has_simple_content_inner(children: &[E4XOrXml<'_>]) -> bool {
+    match children {
+        [] => true,
+        [child] => child.node().has_simple_content(),
+        _ => children
+            .iter()
+            .all(|child| !matches!(&*child.node().kind(), E4XNodeKind::Element { .. })),
+    }
+}
+
+pub fn init<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let this = this.unwrap().as_xml_list_object().unwrap();
+    let value = args[0];
+
+    match E4XNode::parse(value, activation) {
+        Ok(nodes) => {
+            this.set_children(
+                activation.context.gc_context,
+                nodes.into_iter().map(E4XOrXml::E4X).collect(),
+            );
+        }
+        Err(e) => {
+            return Err(Error::RustError(
```

Comment:
We should get into the habit of throwing avm errors where applicable now - there's definitely content that try/catches this